crates/rust-analyzer/src/cli/lsif.rs

```diff
@@ -0,0 +1,260 @@
+//! Lsif generator
+
+use std::collections::HashMap;
+use std::env;
+use std::time::Instant;
+
+use ide::{
+    Analysis, Cancellable, FileId, FileRange, RootDatabase, StaticIndex, StaticIndexedFile,
+    TokenId, TokenStaticData,
+};
+use ide_db::LineIndexDatabase;
+
+use ide_db::base_db::salsa::{self, ParallelDatabase};
+use lsp_types::{lsif::*, Hover, HoverContents, NumberOrString};
```

Comment:
I suggest `use lsp_types::{self, lsif}` so that it's clearer at the call site where the types are comming from.